pkg/util/mount/exec_mount.go

```diff
@@ -0,0 +1,124 @@
+// +build linux
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mount
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+)
+
+// ExecMounter is a mounter that uses provided Exec interface to mount and
+// unmount a filesystem. For all other calls it uses a wrapped mounter.
+type execMounter struct {
+	wrappedMounter Interface
+	exec           Exec
+}
+
+func NewExecMounter(exec Exec, wrapped Interface) Interface {
+	return &execMounter{
+		wrappedMounter: wrapped,
+		exec:           exec,
+	}
+}
+
+// execMounter implements mount.Interface
+var _ Interface = &execMounter{}
+
+// Mount runs mount(8) using given exec interface.
+func (m *execMounter) Mount(source string, target string, fstype string, options []string) error {
+	bind, bindRemountOpts := isBind(options)
+
+	if bind {
+		err := m.doExecMount(source, target, fstype, []string{"bind"})
+		if err != nil {
+			return err
+		}
+		return m.doExecMount(source, target, fstype, bindRemountOpts)
+	}
+
+	return m.doExecMount(source, target, fstype, options)
+}
+
+// doExecMount calls exec(mount <waht> <where>) using given exec interface.
+func (m *execMounter) doExecMount(source, target, fstype string, options []string) error {
+	glog.V(5).Infof("Exec Mounting %s %s %s %v", source, target, fstype, options)
+	mountArgs := makeMountArgs(source, target, fstype, options)
+	output, err := m.exec.Run("mount", mountArgs...)
+	glog.V(5).Infof("Exec mounted %v: %v: %s", mountArgs, err, string(output))
+	if err != nil {
+		glog.Errorf("Mount failed: %v\nMounting command: %s\nMounting arguments: %s %s %s %v\nOutput: %s\n", err, "mount", source, target, fstype, options, string(output))
+		return fmt.Errorf("mount failed: %v\nMounting command: %s\nMounting arguments: %s %s %s %v\nOutput: %s\n",
+			err, "mount", source, target, fstype, options, string(output))
+	}
+
+	return err
+}
+
+// Unmount runs umount(8) using given exec interface.
+func (m *execMounter) Unmount(target string) error {
+	outputBytes, err := m.exec.Run("umount", target)
+	glog.V(5).Infof("Exec unmounted %v: %s", err, string(outputBytes))
+	if len(outputBytes) != 0 {
+		glog.V(5).Infof("Output of unmounting %s: %v", target, string(outputBytes))
```

Comment:
combine this with the above log message

Reply:
reworked a bit to have higher log level on error